nwbwidgets/spectrum.py

```diff
@@ -0,0 +1,59 @@
+from ndx_spectrum import Spectrum
+import warnings
+import matplotlib.pyplot as plt
+from collections.abc import Iterable
+from h5py import Dataset
+from ipywidgets import widgets
+import numpy as np
+from ipywidgets.widgets.interaction import show_inline_matplotlib_plots
+
+
+def show_spectrum(node, **kwargs):
+    if isinstance(node, Spectrum):
+        power = _data_to_array(node.power)
+        if len(power.shape)==2:
+            no_channels = power.shape[1]
+        else:
+            no_channels = 1
+        out = widgets.Output()
+        with out:
+            widgets.interact(lambda channel_no: sp(node, channel_no), channel_no=(0, no_channels, 1))
+            show_inline_matplotlib_plots()
+        return out
```

Comment:
use vis2widget

Reply:
I'm also adding a slider with the plot to show only one channel. `vis2widget `is called by the parent function anyway (if the output is anything other than a plot).
One option that I considered was to only output a plot of multi-channel traces of power in one figure, but the plot wasn't very good to look at,so I added a **slider** (using the widgets.interact) to select only one channel to plot the trace for.

To link a `widget.value` to an input arg to a figure I tried two ways: to slip using the widgets.Output() and only returning a widgets.interact(), but in this case the `show_inline_matplotlib_plots()` was not working during rendering. With out that it correctly shows inline but there are two figures instead of one. Can you try it with the nwb file in the main draft comment?

Reply:
as an update: there are two sliders now: one for channel range and other for frequency range. The double plot showing up problem is solved but I still have kept this as a `widgets.Output()`. vis2widget would not work since it checks for the function returning a plt.plot() object

Reply:
please use utils.widget.interactive_output like it is done here: https://github.com/NeurodataWithoutBorders/nwb-jupyter-widgets/blob/81b8b4f977dc7d40dc4b0f188d783b94f224a4fc/nwbwidgets/misc.py#L110-L117